hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -449,6 +449,10 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
   Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
     String warehouseLocation = conf.get("hive.metastore.warehouse.dir");
 
+    Preconditions.checkNotNull(
+            warehouseLocation,
+            "Warehouse location is not set: hive.metastore.warehouse.dir=null");
```

Comment:
Style: indentation should be 4 spaces (2 indents) from the start of `Preconditions`.